src/dmd/objc.d

```diff
@@ -122,7 +122,7 @@ struct ObjcSelector
         }
     Lcomplete:
         buf.writeByte('\0');
-        return lookup(cast(const(char)*)buf[].ptr, buf.size, pcount);
+        return lookup(cast(const(char)*)buf[].ptr, buf[].length - 1, pcount);
```

Comment:
Why `-1`?

Reply:
The terminating 0 is not needed.

Reply:
Selectors are 0 terminated.

Reply:
Yeah, but the call is to lookup() which then calls stringtable.update() which calls stringtable.update() which calls allocValue() which adds its own 0 termination.

Reply:
OK.

Reply:
Is there a point in doing line 124 then, if `allocValue` adds its own 0 termination?

Reply:
I'll leave that to you.

Reply:
Ok, but I'll leave it to @jacob-carlborg to re-evaluate the logic of the function.